packages/plugin-char/src/VNodes/CharNode.ts

```diff
@@ -0,0 +1,144 @@
+            let next = this.nextSibling();
+            const charNodes: CharNode[] = [this];
+            while (next && this._isSameAs(next)) {
+                if (next instanceof CharNode) {
+                    charNodes.push(next);
+                    if (next.char === ' ' && text[text.length - 1] === ' ') {
+                        // Browsers don't render consecutive space chars otherwise.
+                        text += '\u00A0';
+                    } else {
+                        text += next.char;
+                    }
+                }
+                next = next.nextSibling();
+            }
+            // Browsers don't render leading/trailing space chars otherwise.
+            text = text.replace(/^ | $/g, '\u00A0');
+
+            // Create and append the text node, update the VDocumentMap.
+            let renderedNodes = [document.createTextNode(text)] as Node[];
+            if (this.attributes.size) {
+                this.attributes.forEach(attribute => {
+                    renderedNodes = attribute.render(renderedNodes);
+                });
+            }
+            const fragment = document.createDocumentFragment();
+            renderedNodes.forEach(node => fragment.appendChild(node));
+            return ({ fragment: fragment, vNodes: charNodes } as unknown) as T;
+        } else {
+            return this.renderingEngines[to].render(this) as T;
+        }
+    }
+    /**
+     * Return a new VNode with the same type and attributes as this VNode.
+     *
+     * @override
+     */
+    shallowDuplicate(): CharNode {
+        return new CharNode(this.char);
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    toString(): string {
+        let string = '<' + this.constructor.name + ':' + this.name;
+        if (this.hasChildren()) {
+            string += '>';
+            this.children.forEach(child => {
+                string += child.toString();
+            });
+            string += '</' + this.constructor.name + ':' + this.name + '>';
+        } else {
+            string += '/>';
+        }
+        return string;
+    }
+    /**
+     * Return the length of this VNode.
+     */
+    get length(): number {
+        return 1;
+    }
+    /**
+     * Return this VNode's inner text (concatenation of all descendent
+     * char nodes values).
+     *
+     * @param __current
+     */
+    text(__current = ''): string {
+        __current += this.char;
+        return __current;
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Return true if this VNode has the same format properties as `b`.
+     *
+     * @param b
+     */
+    _isSameAs(b: VNode): boolean {
```

Comment:
Will this still work?

Reply:
It seems to but `<i>’Twas brillig, <b>and</b> the slithy toves</i>` doesn't parse properly: we get `<i>’Twas brillig, </i><b>and</b> the slithy toves` instead.